en/protocol/parameter.md

```diff
@@ -1,7 +1,64 @@
+
+## Supported Data Types
+
+MAVLink v1.0 supports these data types:
+
+* `uint32_t` - 32bit unsigned integer (use the ENUM value [MAV_PARAM_TYPE_UINT32](../messages/common.md#MAV_PARAM_TYPE_UINT32))
+* `int32_t` - 32bit signed integer (use the ENUM value [MAV_PARAM_TYPE_INT32](../messages/common.md#MAV_PARAM_TYPE_INT32))
+* `float` - IEEE754 single precision floating point number (use the ENUM value [MAV_PARAM_TYPE_FLOAT](../messages/common.md#MAV_PARAM_TYPE_FLOAT))
+
+> **Note** All parameters are send as the float value of `mavlink_param_union_t`, which means that a parameter should be byte-wise converted with this union to a byte-wise float (no type conversion). This is necessary in order to not limit the maximum precision for scaled integer params. E.g. GPS coordinates can only be expressed with single float precision up to a few meters, while GPS coordinates in 1E7 scaled integers provide very high accuracy.
+
+```c
+mavlink_param_union_t param;
+
+int32_t integer = 20000;
+
+param.param_int32 = integer;
+param.type = MAV_PARAM_TYPE_INT32;
+
+// Then send the param by providing the float bytes to the send function
+mavlink_msg_param_set_send(xx, xx, param.param_float, param.type, xx);
+```
+
+## Multi-System and Multi-Component Support
+
+MAVLink supports multiple systems / airplanes in parallel on the same link. In addition to this, it also supports multiple MAVLink-enabled devices in the same airplane. The protocol for example allows to communicate over one radio link with the autopilot and a payload unit. For this reason the parameter protocol also differentiates between components. To get a complete parameter list from a system, send the request parameter message with `target_component` set to 0 (enum value: [MAV_COMP_ID_ALL](../messages/common.md#MAV_COMP_ID_ALL)). All onboard components should respond to parameter request messages with their ID or with ID `MAV_COMP_ID_ALL` (0). QGroundControl by default queries all components of a system (it only queries the currently selected system, not all systems) and therefore sends ID 0 / `MAV_COMP_ID_ALL`.
+
+
+## Graphical User Interface in QGroundControl
+
+For this reason the parameter interface *discriminates between systems (one system is one airplane) and components (one component is one entity in the architecture, e.g. the IMU or a Linux process).* This allows to transparently access the individual component parameters without the need of a central onboard unit that translates the parameter read/write requests for the onboard components.
+
+As can be seen on the image below, each component is represented by a top-level node in the parameter tree. The system (the MAV) can be selected in the top-level drop-down menu. The GUI keeps track of changed parameters will send those parameters which changed to the appropriate components.
+
+![QGroundControl Parameter Interface](../../assets/protocols/parameter_interface_gui.png)
```

Comment:
Image is pretty old. If this is still accurate, would be good to get a version from current QGC.

Reply:
That's pretty ancient, yes :smile: